crates/typst-syntax/src/parser.rs

```diff
@@ -398,6 +409,9 @@ fn math_delimited(p: &mut Parser) {
     while !p.at_set(syntax_set!(Dollar, End)) {
         if math_class(p.current_text()) == Some(MathClass::Closing) {
             p.wrap(m2, SyntaxKind::Math);
+            if p.at(SyntaxKind::RightParen) {
+                p.convert(SyntaxKind::Text);
+            }
```

Comment:
We actually can't just convert everything to text here. For example, we could be at the math shorthand `|]` (this is the only case though, afaik), in which case we shouldn't convert it to a `Text` kind.

Reply:
ugh I always forget `|]`. Should definitely add as a comment then. I still don't fully like splitting p.convert_and_eat, especially because this is morally the same thing. Thoughts on just using `else { eat() }`?

Reply:
Yeah I'm fine with adding `else { eat() }` over a new method.